bindings/go/oci/blob/artifact_blob.go

```diff
@@ -193,6 +195,60 @@ func (r *ArtifactBlob) OCIDescriptor() ociImageSpecV1.Descriptor {
 	}
 }
 
+// HasSizeAndDigest checks if the blob has both a pre-calculated size and a pre-calculated  digest.
+func (r *ArtifactBlob) HasSizeAndDigest() bool {
```

Comment:
I think this is not reliable enough. Just because the digest and size are not available, doesn't mean that the artifact blob doesn't expose a digest and/or size, just if it is available as a precomputed variable.

so I think it may be needed to attempt to fetch the digest and or size and if that fails, to compute them with the buffer.

Reply:
I think actually you want something like `ExposesSize` and that can be checked by asserting that the blob implements an interface called [`DigestAware` for digests](https://github.com/open-component-model/open-component-model/blob/main/bindings/go/blob/blob.go#L67-L73) and [`SizeAware`](https://github.com/open-component-model/open-component-model/blob/main/bindings/go/blob/blob.go#L42-L49) for sizes.